pkg/kubelet/dockertools/manager.go

```diff
@@ -1170,6 +1171,9 @@ func (dm *DockerManager) killContainer(containerID types.UID) error {
 					break
 				}
 			}
+			if pod.Spec.TerminationGracePeriodSeconds != nil {
+				gracePeriod = uint(*pod.Spec.TerminationGracePeriodSeconds)
```

Comment:
If we don't want to drive decisions off DeletionTimestamp due to potential clock skew, then we need to record the value of DeleteOptions.GracePeriodSeconds and use that.